app.py

```diff
@@ -111,10 +109,15 @@ def main():
     # TODO -- generate a survey link for a particular day
     # We are awaiting finalization of the REDCap project to know how
     # daily attestations (repeating instruments) will be implemented.
-    if is_complete('eligibility_screening', user_data):
+    if is_complete('eligibility_screening', user_data) and \
+        is_complete('consent_form', user_data) and \
+        is_complete('enrollment_questionnaire', user_data):
         return f"Congrats, {net_id}, you're already registered under record ID " \
             f"{user_data['record_id']} and your eligibility " \
-            "screening is complete!"
+            "screening, consent form, and enrollment questionnaires are complete!"
 
-    # Generate a link to the eligibility questionnaire, and then redirect
+    # Generate a link to the eligibility questionnaire, and then redirect.
+    # Because of REDCap's survey queue logic, we can point a participant to an
+    # upstream survey. If they've completed it, REDCap will automatically direct
+    # them to the next, uncompleted survey in the queue.
     return redirect(generate_survey_link(user_data['record_id']))
```

Comment:
That's really neat. Could we let Redcap handle linking to enrollment vs attestation using survey queue logic as well? (I'm not sure if those all count as the same survey queue though)

Reply:
Wouldn't that be great? I'm not sure yet, but hopefully it can be that simple on our side.

Reply:
Sadly we can't sue the survey queue for repeating instruments, so this won't be as simple for the daily attestations.

Reply:
Too bad. Thanks for the info  @joverlee521!

Reply:
One of the things we determined a couple weeks ago is that we can't use the survey queue for the encounter events (which the daily attestation falls into) because REDCap doesn't sufficiently support survey queue logic for repeating events. This is one reason why Musher exists.

Another problem that Musher solves is creating new instances of the event each day (but only when someone visits), which is something that's not automatable in REDCap otherwise.